Source/utils/str_cat.hpp

```diff
@@ -0,0 +1,80 @@
+#pragma once
+
+#include <cstring>
+#include <string>
+#include <type_traits>
+
+#include "utils/stdcompat/string_view.hpp"
+
+namespace devilution {
+
+/**
+ * @brief Appends the integer to the given buffer.
+ * @return char* end of the buffer
+ */
+char *BufAppend(char *out, int value);
+
+/**
+ * @brief Appends the integer to the given string.
+ */
+void StrAppend(std::string &out, int value);
+
+/**
+ * @brief Appends the given string_view to the given buffer.
+ */
+inline char *BufAppend(char *out, string_view value)

```

Comment:
This writes to the given pointer directly overwriting anything that was already there. The string version actually appends to the string, the BufAppend functions would be clearer with a different name imo.

```suggestion
 * @brief Copies the given string_view to the given buffer.
 *
 * Does not null terminate the resulting c-str, does not truncate if the string_view is larger than the buffer
 */
inline char *BufCopy(char *out, string_view value)
```

Reply:
Done